src/test/java/in/projecteka/consentmanager/link/link/LinkUserJourneyTest.java

```diff
@@ -571,4 +575,117 @@ public void shouldFailOnLinkCareContexts() throws Exception {
+                "  }\n" +
+                "}";
+        var linkReferenceJson = "{\n" +
+                "  \"transactionId\": \"7f7a535d-a3fd-416b-b069-c97d021fbacd\",\n" +
+                "  \"link\": {\n" +
+                "    \"referenceNumber\": \"ref-no\",\n" +
+                "    \"authenticationType\": \"DIRECT\",\n" +
+                "    \"meta\": \n" +
+                "      {\n" +
+                "        \"communicationMedium\": \"M0BILE\",\n" +
+                "        \"communicationHint\": \"test-hint\", \n" +
+                "        \"communicationExpiry\": \"2020-12-30T12:01:55Z\"\n" +
+                "      }\n" +
+                "  }\n" +
+                "}";
+        when(authenticator.verify(token)).thenReturn(Mono.just(new Caller("user-id", false)));
+        gatewayServer.enqueue(new MockResponse().setHeader("Content-Type", "application/json").setBody("{}"));
+        clientRegistryServer.setDispatcher(dispatcher);
+        when(linkRepository.getHIPIdFromDiscovery(patientLinkReferenceRequest.getTransactionId()))
+                .thenReturn(Mono.just(hipId));
+        when(linkRepository.insert(OBJECT_MAPPER.readValue(linkReferenceResult, PatientLinkReferenceResult.class), hipId, patientLinkReferenceRequest.getRequestId()))
+                .thenReturn(Mono.create(MonoSink::success));
+        when(linkRepository.selectLinkReference(patientLinkReferenceRequest.getRequestId()))
+                .thenReturn(Mono.empty());
+        when(linkServiceClient.linkPatientEnquiryRequest(linkReferenceRequest, token, hipId)).thenReturn(Mono.just(true));
+        when(linkResults.get(any())).thenReturn(Mono.just(linkReferenceResult));
+        webTestClient
+                .post()
+                .uri("/v1/links/link/init")
+                .header("Authorization", token)
+                .contentType(MediaType.APPLICATION_JSON)
+                .bodyValue(patientLinkReferenceRequest)
+                .accept(MediaType.APPLICATION_JSON)
+                .exchange()
+                .expectStatus()
+                .isOk()
+                .expectBody()
+                .json(linkReferenceJson);
+    }
+
+    @Test
+    public void shouldFailPatientLinkReference() throws IOException {
+        var token = string();
+        var patientLinkReferenceRequest = patientLinkReferenceRequest().build();
+        var linkReferenceRequest = TestBuilders.linkReferenceRequest().build();
+        var hipId = "10000005";
+        var linkReferenceResult = "{\n" +
+                "  \"requestId\": \"5f7a535d-a3fd-416b-b069-c97d021fbacd\",\n" +
+                "  \"timestamp\": \"2020-05-25T15:03:44.557Z\",\n" +
+                "  \"transactionId\": \"7f7a535d-a3fd-416b-b069-c97d021fbacd\",\n" +
+                "  \"link\": {\n" +
+                "    \"referenceNumber\": \"ref-no\",\n" +
+                "    \"authenticationType\": \"DIRECT\",\n" +
+                "    \"meta\": \n" +
+                "      {\n" +
```

Comment:
Please use the testBuilder instead of string.

Reply:
Done